src/valhalla_run_matrix.cc

```diff
@@ -244,12 +244,16 @@ int main(int argc, char* argv[]) {
   LogResults(optimize, options, res);
 
   // Run with TimeDistanceMatrix
-  TimeDistanceMatrix tdm;
+  const bool forward_search = options.sources().size() <= options.targets().size();
   for (uint32_t n = 0; n < iterations; n++) {
     res.clear();
-    res = tdm.SourceToTarget(options.sources(), options.targets(), reader, mode_costing, mode,
-                             max_distance);
-    tdm.clear();
+    if (forward_search) {
+      res = TimeDistMatrixForward().SourceToTarget(options.sources(), options.targets(), reader,
+                                                   mode_costing, mode, max_distance);
+    } else {
+      res = TimeDistMatrixReverse().SourceToTarget(options.sources(), options.targets(), reader,
+                                                   mode_costing, mode, max_distance);
+    }
```

Comment:
This is what I mean.. The matrix algo should know which one to apply. I'm not sure how to handle this. I'm not sure how to deal with this, and would appreciate some feedback. AFAICT there's 2 options:
1. keep the templated function and introduce another public function (outside of the TimeDistanceMatrix class) which takes the `options` as argument and returns the right algo based on `sources` & `targets`
2. make the class non-templated and only template the functions which need to know about the direction. Also keep `SourceToTarget` non-templated so it can decide what to do based on the arguments (possibly introduce a protected `ComputeMatrix` or so to do what `SourceToTarget` is doing right now)

Opinions?

Reply:
I went for 2., cleaner IMO.